staging/src/k8s.io/client-go/tools/leaderelection/resourcelock/interface.go

```diff
@@ -160,17 +167,23 @@ type Interface interface {
 	// Describe is used to convert details on current resource lock
 	// into a string
 	Describe() string
+
+	// StartSync starts the synchronization of local cache, which is
+	// running in background until the stop channel gets closed.
+	StartSync(stopCh <-chan struct{})
 }
 
 // Manufacture will create a lock of a given type according to the input parameters
-func New(lockType string, ns string, name string, coreClient corev1.CoreV1Interface, coordinationClient coordinationv1.CoordinationV1Interface, rlc ResourceLockConfig) (Interface, error) {
+func New(lockType string, ns string, name string, coreClient corev1.CoreV1Interface, coordinationClient coordinationv1.CoordinationV1Interface, leaseInformer informerv1.LeaseInformer, rlc ResourceLockConfig) (Interface, error) {
```

Comment:
this is changing a signature on a public function, this will force people to have to modify code when revendoring

Reply:
Would that be a big problem? 

Alternatively, we can keep it as it is and create a separate public function, but if users use the current one, there is no optimization. WDYT?

Reply:
as a person that has to deal with revendors I always want us to think if we can avoiding people to have to modify their code, I do not how much is used this function though

Reply:
I did a `grep`, only [one place](https://github.com/kubernetes/dynamic-resource-allocation/blob/v0.28.4/leaderelection/leaderelection.go#L186) uses this `New` func in k8s codebase. But for sure, many non-k8s controllers may use it, which we have no visibility.

Reply:
Make sense. I changed it with adding a new function.